001-IntroToKubernetes/Coach/09-helm.md

```diff
@@ -4,12 +4,71 @@
-- **NOTE:** Helm 3 does not create namespaces with yaml files in the chart.
-	- The YAML files in the Student Resources folder provides a YAML file that creates a whatthehack namespace. 
-	- Students can not include the namespace YAML when creating the helm chart
-	- They have to create the namespace by using the **--create-namespace** flag on the helm CLI, eg:
-		- `helm install myRelease app-languages --create-namespace whatthehack`
-- Consider using the helm chart provided in the Coach Solutions folder for Challenge 9 as one that can be demoed during the lecture.  
+	- At a minimum make sure the version label is parameterized (ie: v1, v2, etc) so that students can install different versions of the app using `helm install` easily.
+- **NOTE:** Helm 3 does not automatically create namespaces.
+	- The YAML files in the Student Resources folder provides a YAML file that creates a whatthehack namespace.
+    	- This should only be used when installing the app manually during the first part of the challenge.
+- **NOTE:** When creating their helm charts make sure that students are handling namespaces properly:
+	- Students should NOT include the namespace YAML in the helm chart
+	- The application's YAML files should NOT include a namespace, we will leave namespace creation up to helm (see below).
+
+## Helm Commands 
+In this section we'll lay out all the commands needed to create the chart using the `app-languages-helm-chart` folder in the Coach Solutions for **Challenge 9** and get it pushed up to your Azure Container Registry
+- First of all we need to allow admin access and authenticate ourselves with the ACR and add it as a helm repository on our local system
+
+  ```bash
+  # enable admin user on ACR
+  az acr update -n myacr -g myrg --admin-enabled true
+
+  # get credentials for Admin user on ACR (save these values)
+  az acr credential show -n myacr -g myrg 
+
+  # login Helm to ACR using admin credentials (use values from above for -u and -p)
+  export HELM_EXPERIMENTAL_OCI=1
+  helm registry login myacr.azurecr.io -u myuser -p mypass
+
+  # Add ACR as Helm Repo and confirm it is in the list
+  az acr helm repo add --name myacr
+  helm repo list
+  ```
+
+- Now we can package up the `app-languages-helm-chart` chart and push it to our ACR helm repo
+	- Packaging creates a tarball using the name of our chart provided in `Chart.yaml`
+  
+  ```bash
+  # package Helm chart
+  helm package app-languages-helm-chart
+
+  # push Helm Chart to ACR
+  az acr helm push -n myacr langfacts-0.1.0.tgz
+  ```
+
+- Now let's update our local repo's index and search to see if the `langfacts` chart is available
+
+  ```bash
+  # update Helm repo
+  helm repo update
+
+  # search for the langfacts chart to make sure it got uploaded
+  helm search repo lang
+  ```
+
+- Finally let's install our new chart and make sure it is in the releases as well as check that everything is deployed in Kubernetes
```

Comment:
Copy this paragraph and the commands under it and PASTE it all back in after the current Line 18.

AFTER you paste a copy of this whole section up on line 18, make the following modifications to the pasted section:
1) Modify "Finally let's install" to "First, let's install"

2) For the helm install command, you will need to remove the parameters dealing with ACR.  This command should be able to install the helm chart from the student's local environment, not ACR.

Leave the original content on lines 59 and down in place.  Yes, you are repeating yourself, but you are explaining how to run the commands & verify it works from ACR.